### DISCUSSION TITLE
Vibration compensation/input shaping?

### DISCUSSION THREAD

**Discussion by RyantHults** [xblax/flashforge_ad5m_klipper_mod#204] [Q&A] [answered]
Does this mod have the ability to tune for vibration compensation? If not, is there anything in particular that's holding that up? I'm eager to get away from the closed Flashforge FW, but really appreciated that feature. I'd be happy to look into implementing that assuming it's not just something I've missed and I can wrap my head around how it works.

**Comment by Avpman2** [comment]
+1 for use case.

**Comment by thhdragon** [comment, marked as answer]
input shaping is standard in klipper yeah. only quirk is you have to test the x and let it compute, then test the y. if you just tell it to test both with shaper_calibrate it starts trying to shake the y before its done calculating the x and it causes klipper to crash because our boards are pretty weak. so just do shaper_calibrate axis=x and then shaper_calibrate axis=y and it will tell you the recommendations in the console

**Reply by consp** [reply]
This is currently the right answer (OP can mark if sufficient). Flashforge does some magic to avoid this problem (like the K1 also does for instance) but that's all proprietary.
